src/deserializer.c

```diff
@@ -47,6 +46,9 @@ AvroDeserializer_init(AvroDeserializer *self, PyObject *args, PyObject *kwds)
     }
     self->flags |= DESERIALIZER_SCHEMA_OK;
 
+    self->schema_json = malloc(strlen(schema_json) + 1);
+    strcpy(self->schema_json, schema_json);
```

Comment:
I'm copying the `schema_json` to become the owner of the `self->schema_json` argument. I'm not sure about the lifecycle of the pointer I get in `PyArg_ParseTuple`\`PyArg_ParseTupleAndKeywords`, it comes from Python side and is deleted somewhere there. If I parse it directly to `self->schema_json`, I'll get an error on `do_close` when trying to free `self->schema_json`. I prefer to claim direct ownership of the arguments I use and to have full control over them.
Also, having this parsed into `self->schema_json` may cause issues when pickling python objects (e.g. spark kafka serde).

Reply:
Fair enough. I'm also not sure of the details of what these pointers are to.  I assume they stick around in some context object that represents the state of the python engine, but who knows.  Safer to make your own copy, since at least then you control its lifetime.